src/components/onboarding/selectable_toggle.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from "react"
+import styled from "styled-components"
+
+import * as fonts from "../../assets/fonts"
+import Icon from "../icon"
+
+interface SelectableToggleProps {
+  href?: string
+  text?: string
+  onSelect: (selected: boolean) => void
+  selected: boolean
+}
+
+const IconContainer = styled.div`
+width: 15px;
+height: 15px;
+background-color: black;
+display: none;
+border-radius: 50%;
+float: right;
+margin-right: 15px;
+`
+
+const Link = styled.a`
+display: block;
+font-size: 14px;
+color: black;
+text-decoration: none;
+text-transform: uppercase;
+font-family: ${fonts.primary.fontFamily};
+padding: 30px 0 30px 15px;
+border-top: 1px solid #e5e5e5;
+&:hover {
+  background-color: #f8f8f8;
+}
+&:hover .collector-intent-checked {
+  display: inline;
+}
+& .collector-intent-checked.is-selected {
+  display: inline;
+}
+`
+
+class SelectableToggle extends React.Component<SelectableToggleProps, null> {
+  constructor(props) {
+    super(props)
+    this.onSelect = this.onSelect.bind(this)
+  }
+
+  onSelect() {
+    if (this.props.onSelect) {
+      this.props.onSelect(this.props.selected)
```

Comment:
I’m not sure I understand this. I assume that the boolean being passed to the `onSelected` callback is supposed to reflect the new `selected` status? How does `this.props.selected` get updated after selecting an option?

Reply:
@alloy the `onOptionSelected()` updates the selection state in the `BudgetComponent` and we are comparing whether the selection string matches the text of the clicked `SelectableToggle` to produce the `selected` prop. In that case, the one selected will be true and the other `SelectableToggle` components will be false. It seems like the BudgetComponent should be the one to know about which option was clicked and just let the SelectableToggle component know if it was selected or not in the form of a prop.